lmdeploy/cli/utils.py

```diff
@@ -358,6 +358,33 @@ def chat_template(parser):
             'Please refer to https://lmdeploy.readthedocs.io/en/latest/advance/chat_template.html for the specification'  # noqa
         )
 
+    @staticmethod
+    def enable_reasoning(parser):
+        """Add argument enable_reasoning to parser."""
+
+        return parser.add_argument('--enable-reasoning',
+                                   action='store_true',
+                                   default=False,
+                                   help='Whether to enable reasoning extraction. Default to disable.')
+
+    @staticmethod
+    def reasoning_parser(parser):
+        """Add reasoning parser to parser."""
+
+        return parser.add_argument('--reasoning-parser',
+                                   type=str,
+                                   default=None,
```

Comment:
can we put `ReasoningParserManager.module_dict.keys()` as choices in here for better help info in cli?